modules/ecs-service/main.tf

```diff
@@ -0,0 +1,696 @@
+# in an existing ECS cluster.
+#
+# The ECS service can support both FARGATE and EC2 compute. In addition to compute,
+# the module supports running the task-definition as both a service or as a scheduled 
+# task that can be triggered by an Event or rule.
+#
+# The module includes the following:
+#
+# - TODO: ADD THIS IN DEVELOPMENT
+#
+# -------------------------------------------------------------------------------------
+
+
+# -------------------------------------------
+# SET TERRAFORM REQUIREMENTS TO RUN MODULE
+# -------------------------------------------
+
+terraform {
+  required_version = ">= 1.5.5"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 5.0"
+    }
+    random = {
+      source  = "hashicorp/random"
+      version = ">= 3.5.1"
+    }
+  }
+}
+
+
+# -------------------------------------------
+# RETRIEVE GENERAL AWS INFORMATION
+# -------------------------------------------
+
+data "aws_caller_identity" "current" {}
+
+data "aws_region" "current" {}
+
+
+# -------------------------------------------
+# RETRIEVE THE ECS CLUSTER INFORMATION
+# -------------------------------------------
+
+data "aws_ecs_cluster" "cluster" {
+  cluster_name = var.ecs_cluster_name
+
+  lifecycle {
+    # If the ECS cluster is not in an ACTIVE state, then the ECS service cannot 
+    # be created and the module should fail.
+    postcondition {
+      condition     = self.status == "ACTIVE"
+      error_message = "The ECS cluster must be in an ACTIVE state before creating the ECS service. Currently the ECS cluster is ${self.status}."
+    }
+
+    # If the ECS service is using Service Connect, a default namespace must be
+    # configured for the ECS cluster.
+    postcondition {
+      condition     = !var.enable_service_connect || self.service_connect_defaults[0].namespace != null
+      error_message = "An ECS service using Service Connect must have a default namespace configured for the ECS cluster."
+    }
+  }
+}
+
+
+# -------------------------------------------
+
+# IDENTIFY THE TASK DEFINITION REVISION TO USE 
+
+# FOR THE ECS SERVICE OR SCHEDULED TASK
+
+# -------------------------------------------
+
+locals {
+  # This allows us to query both the existing as well as Terraform's state and get
+  # and get the max version of either source, useful for when external resources
+  # update the container definition
+  max_task_def_revision = max(aws_ecs_task_definition.task.revision, data.aws_ecs_task_definition.task.revision)
+  task_definition       = "${aws_ecs_task_definition.task.arn_without_revision}:${local.max_task_def_revision}"
+}
+
+# This allows us to query both the existing as well as Terraform's state and get
+# and get the max version of either source, useful for when external resources
+# update the container definition
+data "aws_ecs_task_definition" "task" {
+  task_definition = aws_ecs_task_definition.task.family
+
+  depends_on = [
+    aws_ecs_task_definition.task
```

Comment:
No, this is a data block but it is depending on the resource block.

Since resource creates/maintains the infra that is what allows us to determine config while the data block queries existing infrastructure. This provides us the ability to wait to query only after the resource is created. This allows for bootstrapping of the image but after it is initialized it can query for external changes to the revision which allows us to use the latest revision all the time. @mfranz1